pkg/codec/xml/array.go

```diff
@@ -31,26 +30,62 @@ func NewArray(name string, template specs.Template, repeated specs.Repeated, ref
 // MarshalXML encodes the given specs object into the provided XML encoder.
 func (array *Array) MarshalXML(encoder *xml.Encoder, _ xml.StartElement) error {
 	if array.reference == nil {
+		for _, item := range array.repeated {
+			if err := encodeElement(encoder, array.name, item, array.store); err != nil {
+				return err
+			}
+		}
+
 		return nil
 	}
 
 	var reference = array.store.Load(array.reference.Resource, array.reference.Path)
-	if reference == nil {
+	if reference == nil || reference.Repeated == nil {
+		// ignore
 		return nil
 	}
 
-	if reference.Repeated == nil {
-		return errors.New("reference does not contain repeated value")
-	}
+	array.template.Reference = new(specs.PropertyReference)
```

Comment:
I think in the end we will have to move this implementation to the providers (ex: HCL)

Reply:
impossible :disappointed: 
then the reference will always be set, we need to differentiate:
- nil reference
- empty reference (with no resource and path) e.g. `"",""`
- filled reference (`"input","path"`)